Databaseinstantiation

Lindayezou opened the pull request:
Database will be populated with game, user, gymleader and pokemon data when server starts.

Commit:
Creates data file structure

Creates data folder and data files for gym leader, pokemon, users, and gameboard

 Changes to be committed:
	new file:   data/gameBoardData.js
	new file:   data/gymLeaderData.js
	new file:   data/pokemonData.js
	new file:   data/usersData.js

Commit:
Changes hidden

Changes hidden key to visible as hidden is a javascript/angular key word

 Changes to be committed:
	modified:   pokemon/pokemonModel.js

Commit:
Adds 2 gym leaders

Adds two values to the gym leader data, Brock and Misty

 Changes to be committed:
	modified:   data/gymLeaderData.js

Commit:
Adds game board data

Adds all of the node information for the gameboard including, color, type, gymleader

 Changes to be committed:
	modified:   data/gameBoardData.js

Commit:
Adds pokemon data

Adds the data for the 6 starter pokemon: Squirtle, Charmander, Bulbasaur, Pikachu, Clefairy, Meowth

 Changes to be committed:
	modified:   data/pokemonData.js

Commit:
Adds sample user

Adds sample user named Henry with a fake facebook id

 Changes to be committed:
	modified:   data/usersData.js

Commit:
Allow database instantiation with pre-existing data

Database will be populated with game, gymleader and pokemon data upon server start-up
- modified data files to allow export
- changed gymleader and pokemon schemas to only allow one object input
- creates and saves new data only is it does not yet exists
- remove data to allow creating new content

 Changes to be committed:
	modified:   server/data/gameBoardData.js
	modified:   server/data/gymLeaderData.js
	modified:   server/data/pokemonData.js
	modified:   server/data/usersData.js
	modified:   server/db/db.js
	modified:   server/gymLeader/gymLeaderModel.js
	modified:   server/pokemon/pokemonModel.js
	modified:   server/server.js

Commit:
Comments out user requirement

Allow for code to run without error

 Changes to be committed:
	modified:   server/db/db.js

Commit:
Resolves user schema/create collection error

Refactors user schema to have corrent variable names
Implement create user collection logic in db.js

 Changes to be committed:
	modified:   server/db/db.js
	modified:   server/users/userModel.js

Commit:
Corrects id from caps to camelcase

Changes ID keys for items and events schema to be Id

 Changes to be committed:
	modified:   server/events/eventsModel.js
	modified:   server/items/itemsModel.js

hiteshlala commented:
I agree with most of Alex comments, but some of it is not clear to me. - inline comments from now on.
The remove function would be useful during testing to make sure we can control the environment.  Or just delete the the database and recreate it.

updated the function you created your pokemon you have 'attachstrengh' instead of attackstrength

This was a lot of work thank you.

Lindayezou commented:
Thanks for all of the comment, guys. The remove functions are for testing to make sure updates didn't mess up the collections. I will look in to making arrays of objects a bit more and make some changes before tomorrow.

Commit:
Refactors createGymLeader function

Function call now imports each element in the gym leader data file individually instead of one large object

 Changes to be committed:
	modified:   server/db/db.js

Commit:
Refactors createPokemon function

Function call now imports each pokemon data individually instead of one large object

 Changes to be committed:
	modified:   server/db/db.js

Commit:
Updates data files

Removes external number keys for objects

 Changes to be committed:
	modified:   server/data/gymLeaderData.js
	modified:   server/data/pokemonData.js
	modified:   server/data/usersData.js

Commit:
Refactors schemas

Changes the schemas to accommodate importing individual objects

 Changes to be committed:
	modified:   server/gymLeader/gymLeaderModel.js
	modified:   server/pokemon/pokemonModel.js

Commit:
Fixes typo

Changes attach to attack in pokemonData

 Changes to be committed:
	modified:   server/data/pokemonData.js
